Doc/tools/extensions/pyspecific.py

```diff
@@ -147,8 +157,48 @@ def run(self):
         pnode.extend(n + m)
         n, m = self.state.inline_text(self.arguments[0], self.lineno)
         pnode.extend(n + m)
+        if self.content:
+            content = " " + " ".join(self.content)
+            n, m = self.state.inline_text(content, self.content_offset)
+            pnode.extend(n + m)
+
+        self.parse_platforms()
+
         return [pnode]
 
+    def parse_platforms(self):
+        """Parse platform information from arguments
+
+        Arguments is a comma-separated string of platforms. A platform may
+        be prefixed with "not " to indicate that a feature is not available.
+
+        Example::
+
+           .. availability:: Windows, Linux >= 4.2, not Emscripten, not WASI
+
+        """
+        platforms = {}
+        for arg in self.arguments[0].rstrip(".").split(","):
+            arg = arg.strip()
+            platform, _, version = arg.partition(" >= ")
+            if platform.startswith("not "):
+                version = False
+                platform = platform[4:]
+            elif not version:
+                version = True
+            platforms[platform] = version
+
+        unknown = set(platforms).difference(self.known_platforms)
+        if unknown:
+            logger = logging.getLogger(type(self).__qualname__)
+            logger.warn(
+                f"Unknown platform(s) '{' '.join(sorted(unknown))}' in "
+                f"'.. availability:: {self.arguments[0]}'."
```

Comment:
Please add a comment about what to do if this test faills. Devs can just add a platform to the list if it's missing, right?

Reply:
Done